app/src/main/java/com/hover/stax/transactions/TransactionDetailsFragment.kt

```diff
@@ -202,24 +175,47 @@ class TransactionDetailsFragment : DialogFragment() {
                 binding.transactionDetailsCard.setTitle(
                     transaction.generateLongDescription(viewModel.action.value, viewModel.contact.value, requireContext())
                 )
-            binding.infoCard.root.visibility = GONE
         }
 
+        setDetailsData(transaction)
+        setVisibleDetails(transaction)
+        updateDetailsRequiringAction(viewModel.action.value, viewModel.transaction.value)
+        updateStatus(viewModel.action.value, transaction)
+    }
+    private fun setDetailsData(transaction: StaxTransaction) {
+        if(shouldShowNewBalance(transaction)) {
+            binding.primaryStatus.newBalance.apply {
+                text = getString(R.string.new_balance, transaction.displayBalance)
+                visibility = VISIBLE
+            }
+        }
         binding.infoCard.detailsRecipientLabel.setText(if (transaction.transaction_type == HoverAction.RECEIVE) R.string.sender_label else R.string.recipient_label)
         binding.infoCard.detailsAmount.text = transaction.displayAmount
         binding.infoCard.detailsDate.text = humanFriendlyDateTime(transaction.updated_at)
         binding.infoCard.detailsServiceId.text = transaction.confirm_code
         binding.infoCard.detailsStaxUuid.text = transaction.uuid
+        binding.infoCard.detailsStaxStatus.apply {
+            text = transaction.fullStatus.getTitle(requireContext())
+            setCompoundDrawablesWithIntrinsicBounds(0, 0, transaction.fullStatus.getIcon(), 0)
+        }
 
-        setVisibleDetails(transaction)
-        updateStatus(viewModel.action.value, transaction)
+        binding.infoCard.detailsStaxReason.text = transaction.fullStatus.getReason()
+        if(transaction.fee !=null && transaction.fee > 0) binding.infoCard.detailsFee.text = Utils.formatAmount(transaction.fee)
```

Comment:
Still show fee so long as it is not null even if it is 0

Reply:
Done